### ISSUE TITLE
Publish docker image to docker hub

### ISSUE THREAD

**Issue by austinmatherne-wk** [Arelle/Arelle#399]
### What should we change and why?

Similar to the [python package workflow](https://github.com/Arelle/Arelle/blob/34d846eff1b3df8c79dbf95dd86325eb4ca1892b/.github/workflows/python-package.yml#L32-L37) we should add a (tag guarded) job to the workflow created in #398 to publish the docker image to docker hub on release events (will require release environment secrets to handle docker hub authentication).

**Comment by sagesmith-wf** [comment]
What does the image do when started? Does it kick into the webserver or something? Or require some sort of startup config to indicate which mode?

**Comment by austinmatherne-wk** [comment]
It [wraps the CLI](https://dteslya.engineer/blog/2022/07/14/how-to-run-a-python-cli-tool-inside-a-docker-container/#running-the-app-inside-a-container) so you can  `docker run arelleCmdLine --help` and so on. I think we can run the web server from the same image, but if needed we can provide multiple images.

**Comment by mrbrianevans** [comment]
Please could you provide an indication of whether this is still being considered, or if its being worked on when it might be published. Thank you. I think it will be very helpful for getting started with Arelle and deploying it to have a standard image on docker hub.

**Comment by austinmatherne-wk** [comment]
There's [a basic dockerfile in the root of the repo](https://github.com/Arelle/Arelle/blob/master/Dockerfile), but this isn't a top priority at the moment. We're happy to accept contributions if folks in the community would like to push this forward.

**Comment by yipfram** [comment]
Hello @austinmatherne-wk , that's an old issue, but i saw that the install.md document was updated 3 weeks ago, yet the docker hub image is still empty (https://hub.docker.com/r/arelleproject/arelle/tags)
This GH action has no runs https://github.com/Arelle/Arelle/actions/workflows/publish-docker.yml

is it because there is still a need for fixes on the pipline?

Thanks ;)

**Comment by austinmatherne-wk** [comment]
Yes. A couple of items had to be fixed. This should be the last one: https://github.com/Arelle/Arelle/pull/2330 Once that's merged (probably Monday) the docker hub images should be published as part of all future releases.
